hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -3952,17 +3952,27 @@ private void getProcedureResult(long procId, CompletableFuture<Void> future, int
 
   @Override
   public CompletableFuture<List<SlowLogRecord>> getSlowLogResponses(
-    @Nullable final Set<ServerName> serverNames,
+      @Nullable final Set<ServerName> serverNames,
     final SlowLogQueryFilter slowLogQueryFilter) {
     if (CollectionUtils.isEmpty(serverNames)) {
       return CompletableFuture.completedFuture(Collections.emptyList());
     }
-    return CompletableFuture.supplyAsync(() -> serverNames.stream()
-      .map((ServerName serverName) ->
-        getSlowLogResponseFromServer(serverName, slowLogQueryFilter))
-      .map(CompletableFuture::join)
-      .flatMap(List::stream)
-      .collect(Collectors.toList()));
+    if (slowLogQueryFilter.getType() == null
```

Comment:
Why this branch on the client? Can't the server side infer the type and respond accordingly?

Reply:
Nope, both can't use same enum because at server side, we have enum in TooSlowLog.proto which we save in Queue, whereas this enum is for client side filter object.

Having this branch on client side is better compared to passing it on to server because otherwise we will need to have it in Admin.proto too to pass it over in RPC call (too many places).